### ISSUE TITLE
Error during inference/simplification of protein groups with large database

### ISSUE THREAD

**Issue by ohickl** [compomics/peptide-shaker#369]
Hi,
I keep getting the following error message while loading my SearchGUI searches into PeptideShaker:
> Tue Jul 23 08:29:50 CEST 2019: PeptideShaker version 1.16.42.
Memory given to the Java virtual machine: 320691240960.
Total amount of memory in the Java virtual machine: 163053568.
Free memory: 136266080.
Java version: 1.8.0_201.
1714 script command tokens
(C) 2009 Jmol Development
Jmol Version: 12.0.43  2011-05-03 14:21
java.vendor: Oracle Corporation
java.version: 1.8.0_201
os.name: Windows 10
memory: 76.5/231.2
processors available: 28
useCommandThread: false
<CompomicsError>PeptideShaker processing failed. See the PeptideShaker log for details.</CompomicsError>
java.lang.NullPointerException
at com.compomics.util.experiment.identification.protein_sequences.SequenceFactory.getProtein(SequenceFactory.java:389)
at com.compomics.util.experiment.identification.protein_sequences.SequenceFactory.getProteinSynchronized(SequenceFactory.java:333)
	at com.compomics.util.experiment.identification.protein_sequences.SequenceFactory.getProtein(SequenceFactory.java:232)
at com.compomics.util.experiment.identification.protein_sequences.SequenceFactory.getProtein(SequenceFactory.java:198)
at com.compomics.util.experiment.identification.matches.ProteinMatch.hasEnzymaticPeptide(ProteinMatch.java:448)
at eu.isas.peptideshaker.utils.IdentificationFeaturesGenerator.checkEnzymaticPeptides(IdentificationFeaturesGenerator.java:1838)
at eu.isas.peptideshaker.utils.IdentificationFeaturesGenerator.hasEnzymaticPeptides(IdentificationFeaturesGenerator.java:1808)
at eu.isas.peptideshaker.protein_inference.ProteinInference.compareMainProtein(ProteinInference.java:778)
at eu.isas.peptideshaker.protein_inference.ProteinInference.retainBestScoringGroups(ProteinInference.java:545)
at eu.isas.peptideshaker.PeptideShaker.processIdentifications(PeptideShaker.java:448)
at eu.isas.peptideshaker.fileimport.FileImporter$IdProcessorFromFile.importFiles(FileImporter.java:547)
at eu.isas.peptideshaker.fileimport.FileImporter$IdProcessorFromFile.doInBackground(FileImporter.java:431)
at javax.swing.SwingWorker$1.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at javax.swing.SwingWorker.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
Free memory: 23417363880

The database is about 9.5 gb with about 37*10^6 entries.
I used X!Tandem, MyriMatch, Omssa, Andromeda and the de novo sequencing algorithms as well as Mascot. 
I tried to load them all together as well as only singe search engine results. The result was the same. It stops at the simplification of protein groups, I think.
Edit: I already turned the "simplify protein groups" function off in an attempt to fix the problem, so its probably not the cause.

Cheers 

Oskar 

PS 
Love SearchGUI and PeptideShaker! Keep up the amazing work!
Looking forward for PeptideShakers ability to load multiple samples at once, too.

**Comment by hbarsnes** [comment]
Hi Oskar,

This looks like an issue with the use of multiple threads that we have for some time been trying to fix by redesigning this part of our code. A beta version should soon be available for testing. I will keep you posted.

In the meantime you can try using just one thread/core when loading the data in PeptideShaker. This will make the loading much slower, but should hopefully enable you to load the data. You can set this from the New Project dialog via the Processing option found at the bottom of the dialog. Simply change the number of cores to 1.

Best regards,
Harald

**Comment by 0ssH** [comment]
I will try that. Thanks!

**Comment by 0ssH** [comment]
Hi Harald,

I tried loading the data using both the command line version and the gui. I set -threads to 1 or the number of cores to 1 in the gui. Unfortunately I get the same result. Also it always still seems to use all available cores anyway. Am I missing something?

Best regards 

Oskar

**Comment by hbarsnes** [comment]
Hi Oskar,

You may not be able to actually see the result of telling PeptideShaker to use less cores in the task manager or similar, as this setting mainly controls how many threads we use when processing the data in parallel (and thus not really setting the number of cores). Reducing the number of threads should slow down the processing though. Don't you see that happening either?

The good news is that we are working on a new version of PeptideShaker that should solve all such multithreading issues. We hoped to have a beta version ready before the summer holidays, but were not able to sort out all of the remaining issues in time. September therefore seems more likely.

I'll let you know when the beta version becomes available for testing.

Best regards,
Harald

**Comment by 0ssH** [comment]
Got it.
It did not seem to slow down noticeably. But I also did not perform a very scientific check, so it could just be me not noticing.
Looking forward to the next release!
In general, can you remember problems occuring when processing searches with large databases (beyond requiring massive amounts of memory)? The one I am using contains about 18*10^6 target sequences. That is probably not a great starting condition.
Thanks for your time!

**Comment by hbarsnes** [comment]
> In general, can you remember problems occuring when processing searches with large databases (beyond requiring massive amounts of memory)? The one I am using contains about 18*10^6 target sequences. That is probably not a great starting condition.

I don't think we've had anyone using databases that big before. However I don't think the size of the database is the source of the problem. But I could be wrong. Would it be possible for you to share the SearchGUI output (including the FASTA and mgf files) so that we can try loading the data in the upcoming beta release? Hopefully the issue has already been solved.

**Comment by 0ssH** [comment]
Sure, where should I send the link to?

**Comment by hbarsnes** [comment]
> Sure, where should I send the link to?

You can send it to me at harald.barsnes@gmail.com.

**Comment by hbarsnes** [comment]
This issue should hopefully be solved in the new beta releases of SearchGUI and PeptideShaker. Please see https://groups.google.com/forum/#!topic/peptide-shaker/1ecY0IyMOBM for more details.

If the issue still exists in the beta releases, please open a new issue at https://github.com/compomics/peptide-shaker-2.0-issue-tracker/issues.

If the issue has been solved, it would also be great of you could let us know by replying to this original issue and letting us know. :)

**Comment by hbarsnes** [comment]
This issue should hopefully be fixed in the new release that is available now. If not, please let us know by opening a new issue.
